src/inmanta/loader.py

```diff
@@ -28,21 +26,20 @@
 import shutil
 import sys
 import types
-from collections import abc
+from collections import abc, defaultdict
 from collections.abc import Iterable, Iterator, Sequence
 from dataclasses import dataclass
 from importlib.abc import FileLoader, MetaPathFinder
 from importlib.machinery import ModuleSpec, SourcelessFileLoader
 from itertools import chain
-from typing import TYPE_CHECKING, Optional
+from typing import Optional
 
 from inmanta import const, module
+from inmanta.data.model import InmantaModuleDTO, ModuleSource, ModuleSourceMetadata
+from inmanta.resources import Id, Resource
```

Comment:
This makes this module less stand-alone, i.e. more prone to import loops. Might be fine, I'll need to have another look at this.